### PR TITLE
Clean up FixPlrWalkTags

### PR DESCRIPTION
**Pull request by AJenbo**
Bix exact

### CODE REVIEW

File: Source/player.cpp [hunk 1]
```diff
@@ -1688,66 +1688,25 @@ void __fastcall StartSpell(int pnum, int d, int cx, int cy)
...
-					v6 = v3 + 1;
-					v11 = __OFSUB__(v14 + 1, v3 + 1);
-					v9 = v14 + 1 == v3 + 1;
-					v10 = v14++ - v3 < 0;
-				}
-				while ( (unsigned char)(v10 ^ v11) | v9 );
+	}
+
+	int pNext = pnum + 1;
+	int pPrev = -1 - pnum;
```
Comment (squidcc) [inline Source/player.cpp]: This smells like a weird way to get ~pnum. Maybe somebody's keyboard didn't have a tilde key.

Comment (AJenbo) [reply Source/player.cpp]: As I understand it ~pnum would result in -2147483643 to -2147483646, -1 - pnum gives you -2 to -5. In any case they don't produce the same ASM.

Comment (squidcc) [reply Source/player.cpp]: Not sure where you're getting those numbers from, ~ simply flips all the bits.
It's formally defined as an unsigned subtraction from the largest possible value (0xFFFFFFFF i.e. -1).

If it had been used in the original code the compiler would definitely emit a not opcode, this just looks like a different method to get the same result.

Comment (AJenbo) [reply Source/player.cpp]: It's a signed value, does ~ not flip the signed bit?

Comment (squidcc) [reply Source/player.cpp]: It flips all bits including the sign bit.

Comment (seritools) [reply Source/player.cpp]: How about `-(pnum + 1)`, or even `-pNext`? Could still be bin exact :)

Comment (mewmew) [reply Source/player.cpp]: I prefer -1 - pnum. That's what I've been using in my local notes at least ^^

Comment (seritools) [reply Source/player.cpp]: Dunno, `-pnum - 1` just clicks with me better, but maybe that's just me :P

Comment (AJenbo) [reply Source/player.cpp]: I like -pnum - 1 :)

Comment (galaxyhaxz) [reply Source/player.cpp]: Assertion strings:
```
dMonster[monster[i]._mx][monster[i]._my] == i+1
|| dMonster[monster[i]._mx][monster[i]._my] == -(i+1)
```

Comment (seritools) [reply Source/player.cpp]: knew it :P

Comment (mewmew) [reply Source/player.cpp]: Assert to the rescue!

**Pull request merged**